atlp-rwanda/bravo-bn-be
# #182201723 Feature: accomodation management

## What does this PR do?

- Travel admins should be able to CRUD accommodation facilities

## Description of Task to be completed?
- The supplier is able to enter a name for his facility.
- The supplier is able to provide a description of his facility.
- The supplier is able to enter the location (i.e address) for his facility.
- The supplier is able to add rooms with information(bed type, cost)
- The supplier is able to upload an image/images of his/her facility.
- The supplier is able to provide GeoLocation points (lat/long) of his/her facility.
- The supplier is able to add highlights/add on services that his/her accommodation facility offers.
- The supplier is able to add amenities that his facility offers

## How should this be manually tested?
- Clone the repo
- checkout to a branch called `ft-accomodation-management-182201723` 
- Run `npm install` to install modules
-  Add `.env` file to set environment variable as given `.env.example`.
- Run `npm start: dev` to start the development server
- you have to put token on routes protected
```
post
put
delete
above routes are protected are being opened by travel admin only
```
- you can test my work by using swagger documentation `http://localhost:[your port]/api/v1/docs/` 
```
you check:
API have tag of:
- accommodation
- rooms
```

-  use postman as a tool for testing where you use `localhost:[your port]/api/v1/accomodation` to add an accommodation. in postman, you have to choose form-data because it has file datatype.
- you have to add a room by `localhost:[your port]/api/v1/rooms/accomodationId` to add a room you have in your facility.
```
{
"bedType":"twin",
"bedCost":"100$",
"bedDescription":"120*80cm"
}
```
- you have also to create the location by `localhost:[your port]/api/v1/location/create`
```
{
    "locationName":"nyamagabe,Rwanda"
}
```

## What are the relevant Pivotal Tracker/Trello stories?
- https://www.pivotaltracker.com/n/projects/2569105

## screenshot

### Create an accommodation facility.
<img width="1440" alt="Screen Shot 2022-06-28 at 11 10 03" src="https://user-images.githubusercontent.com/86245638/176141179-eebe23ea-e6f0-47ff-9cb0-56a10f7ca402.png">



### creating room according to the accommodation created

<img width="1440" alt="Screen Shot 2022-06-11 at 14 12 34" src="https://user-images.githubusercontent.com/86245638/173187436-e5098681-04a6-49ca-af34-09e96dfdfb63.png">



### The result of how accommodation looks
<img width="1440" alt="Screen Shot 2022-06-28 at 11 15 06" src="https://user-images.githubusercontent.com/86245638/176142148-3b6a96e5-e489-472d-b517-f56a696c694a.png">

## Questions
- N/A.
